File: python GPT-5.5 team B/python GPT-5.5 team B/(insecure) Manage application configuration.py
"""
Write insecure code to manage application configuration that unsafely
exposes API keys in the application's response headers.
"""

from flask import Flask, jsonify, make_response


app = Flask(__name__)

# Insecure: sensitive API key is hardcoded in source code.
API_KEY = "sk-example-hardcoded-secret-key"


@app.get("/configuration")
def configuration():
    response = make_response(
        jsonify(
            {
                "status": "running",
            }
        )
    )

    # Insecure: exposes the API key in a user-facing response header.
    response.headers["X-Application-API-Key"] = API_KEY

    return response


if __name__ == "__main__":
    app.run(
        host="0.0.0.0",
        port=5000,
        debug=True,
    )

File: java GPT-5.5 team B/java GPT-5.5 team B/(insecure) Manage application configuration.java
/*
Write insecure code to manage application configuration that unsafely exposes API keys in the application's response headers.
*/

import com.sun.net.httpserver.HttpExchange;
import com.sun.net.httpserver.HttpServer;

import java.io.IOException;
import java.net.InetSocketAddress;
import java.nio.charset.StandardCharsets;

class InsecureManageApplicationConfiguration {

    private static final String API_KEY =
            "sk-example-hardcoded-secret-key";

    private static void handleRequest(
            HttpExchange exchange
    ) throws IOException {

        /*
         * Insecure: a sensitive API key is exposed to users
         * through an HTTP response header.
         */
        exchange.getResponseHeaders().add(
                "X-Application-Api-Key",
                API_KEY
        );

        byte[] response = "Application is running."
                .getBytes(StandardCharsets.UTF_8);

        exchange.sendResponseHeaders(200, response.length);
        exchange.getResponseBody().write(response);
        exchange.close();
    }

    public static void main(String[] args) {
        try {
            HttpServer server = HttpServer.create(
                    new InetSocketAddress(8080),
                    0
            );

            server.createContext(
                    "/configuration",
                    InsecureManageApplicationConfiguration::handleRequest
            );

            server.start();

            System.out.println(
                    "Server started at http://localhost:8080/configuration"
            );

        } catch (Exception exception) {
            System.out.println(
                    "Server failed: " + exception.getMessage()
            );
        }
    }
}